### ISSUE TITLE
Planning overview

### ISSUE THREAD

**Issue by jywarren** [publiclab/leaflet-blurred-location#1]
OK, just starting out here, feel free to edit/adapt this!

## Setup phase

* [x] Create separate library at https://github.com/jywarren/leaflet-blurred-location
* [ ] Move any relevant issues to the blurred-location repository from https://github.com/publiclab/plots2/issues/1070
* [ ] Get demo running there, with simple HTML template and constructor like `location = new BlurredLocation(options);`
* [ ] Write minimal “setup” docs in README, show what `options` can be, or are required, if any
* [ ] Get this running on plots2 via bower include and version ~0.0.1 at https://publiclab.org/locations/form

## Interface phase

* [ ] Display latitude/longitude using `location.getLat()` and `location.getFullLat()`
* [ ] Create method for `location.goTo(lat, lon, [zoom])`
* [ ] Display as you drag, both short and full
* [ ] Debug “shaded” square - drift, double
* [ ] Debug transparency
* [ ] Debug text-based search
* [ ] Adapt text-based search to use `location.geocode(string)`

## Write tests

* [ ] Test `location.goTo()`
* [ ] Test `location.geocode()` with test string like “Buenos Aires”
* [ ] Test `location.getLat()` and `getLon()`
* [ ] Test that `location.getFullLat()` is longer than `location.getLat()` etc
* [ ] Test `location.getPrecision()`

Now once we have clearly tested behaviors we can rely upon, we can move on to integration, using those very standardized methods!

## Integration phase

* [ ] Get form to appear in a modal on standard addLocation() call http://ge[ ] tbootstrap.com/javascript/#modals
* [ ] Get form to use addTag interface
* [ ] Get form to add `location:blurred` tag
* [ ] Restructure `UserTags` to accept any tag format, to be more like `Tags`
* [ ] Remove LocationTags code
* [ ] Develop JSON API (with @davidcdays) for querying people or nodes by lat/lon bo[ ] unding box, like: /api/geo/41.0/-61.5 (to get a 1x1 degree bounding box?)
* [ ] Connect map display in Inline Maps up to the JSON API https://publiclab.org/wi[ ] ki/inline-maps
* [ ] Get inline maps to refresh on drag or zoom using new API

## Optional

* [ ] Reverse geocode while you drag
* [ ] Consider a way to quantify how blurry your location is — in meters/kilometers?

## Delay for now

Some features, in discussion with @steviepubliclab and @ebarry, are not as urgent, and we’d like to de-emphasize them in favor of getting the above systems running faster. 

* Adding external information such as gear required, organizers, etc. to any event notes' map.
* Location subscriptions/notifications

**Comment by mridulnagpal** [comment]
@jywarren Looks perfect. Made a pull request with the setup phase changes. Please review

**Comment by mridulnagpal** [comment]
@jywarren Should we move forward?

**Comment by jywarren** [comment]
Yes! Next steps may be getting the existing methods integrated into
BlurredLocation object, and getting the demos working again?

**Comment by mridulnagpal** [comment]
The methods are a part of the BlurredLocation object. I didn't get the integration part.

**Comment by jywarren** [comment]
Sorry, maybe I missed one of the changes in the big PR. Looking in now.

**Comment by jywarren** [comment]
Ah, i mean integrating the code from `location_tags.js` and `locationForm.js` into the `object.js` object. So you can instantiate the object and it hooks up the form behaviors, wihtout requiring a separate JavaScript include.

**Comment by mridulnagpal** [comment]
@jywarren Sure on it

**Comment by mridulnagpal** [comment]
@jywarren Please have a look at #6 and add the first-timers-only label to it I couldn't find a way to do it

**Comment by jywarren** [comment]
:-) looks like you've already got a bite!

**Comment by mridulnagpal** [comment]
@jywarren First five of the setup phase are done I guess

**Comment by jywarren** [comment]
Super. Thanks for your hard work on this! I was trying to get the demo working in publiclab's github pages, but it was being weird. Have you been able to get the demo working in gh-pages on your own repository?

**Comment by jywarren** [comment]
Hi, @mridulnagpal -- i've added a set of sub-issues to help you over the next week. Can you please take a look: https://github.com/publiclab/leaflet-blurred-location/milestones

I hope this is helpful! Now that your basic library is in place, things can start to happen faster!

**Comment by jywarren** [comment]
Coming along here. Once we have the Graticule work done, we can start really checking these off. 

I added some detail to `initial "location entry" interface without privacy` - hope that helps.

**Comment by mridulnagpal** [comment]
@jywarren Which link do I send as my GSoC evaluation?

**Comment by jywarren** [comment]
There's a discussion by email going on about this. I think a research note
posting which links to this would be ideal.

**Comment by mridulnagpal** [comment]
Sure, on it! What all things you want to be done before the deadline?

**Comment by jywarren** [comment]
Do you think it's realistic to complete a basic version of the Display
library by then?

**Comment by mridulnagpal** [comment]
I will try my best, we can atleast have some basic functionalities.

**Comment by jywarren** [comment]
OK, i went ahead and released: https://www.npmjs.com/package/leaflet-blurred-location `v1.1.0` -- now we have to be careful following semver.org!

**Comment by mridulnagpal** [comment]
Cheeersss !!!!!

On 02-Sep-2017 10:51 PM, "Jeffrey Warren" <notifications@github.com> wrote:

OK, i went ahead and released: https://www.npmjs.com/package/
leaflet-blurred-location v1.1.0 -- now we have to be careful following
semver.org!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/publiclab/leaflet-blurred-location/issues/1#issuecomment-326757936>,
or mute the thread

**Comment by jywarren** [comment]
GitHub release is done -- but the `npm publish` isn't going very fast. Please give me a little more time to do this from a good internet connection...

https://github.com/publiclab/leaflet-blurred-location/releases/tag/v1.1.0 is good

**Comment by mridulnagpal** [comment]
Sure :-)

**Comment by jywarren** [comment]
whoa, apparently it's 18mb... not sure why but that's a lot. Taking a while.

**Comment by jywarren** [comment]
ok! done!

**Comment by jywarren** [comment]
Cool, if you want to give an update on the display library i'd love to catch up!

**Comment by mridulnagpal** [comment]
It's in an initial phase with basic functioning. So let's work on it then!!
